### PR TITLE
docs: get-logs proposal

### PR DESCRIPTION
**Pull request by jadarsie**
Proposal to discuss features, scope and command interface.

### CODE REVIEW

File: docs/design/get-logs.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+```
+aks-engine get-logs
+    --location ${AZURE_LOCATION}
+    --api-model _out/prefix/apimodel.json
+    --resource-group prefix
+    --output-directory _out/prefix/logs_${STAMP}
+    --client-id ${AZURE_CLIENT_ID}
+    --client-secret ${AZURE_CLIENT_SECRET}
+    --subscription-id ${AZURE_CLIENT_SUBSCRIPTION}
+    --private-key ~/.ssh/id_rsa
```
Comment (jackfrancis) [inline docs/design/get-logs.md]: let's make this `--ssh-private-key`

File: docs/design/get-logs.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+## High level description
+
+Just for consistency, the command interface could be similar to `rotate-certs`’ interface:
+
+```
+aks-engine get-logs
+    --location ${AZURE_LOCATION}
+    --api-model _out/prefix/apimodel.json
+    --resource-group prefix
+    --output-directory _out/prefix/logs_${STAMP}
```
Comment (jackfrancis) [inline docs/design/get-logs.md]: Is this where to put the logs locally?

File: docs/design/get-logs.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    --private-key ~/.ssh/id_rsa
+    --apiserver prefix.location.cloudapp.azure.com
+```
+
+After a successful execution, the output directory could contain this non-exhaustive list of files:
+
+- Output of kubectl cluster-info dump
+- State of the Azure resources in the resource group    
+- For each host
+  - Files in directory /var/log/azure
```
Comment (jackfrancis) [inline docs/design/get-logs.md]: Let's add files under `/var/log` as well

File: docs/design/get-logs.md (hunk 2 above)
Comment (jadarsie) [reply docs/design/get-logs.md]: right

**Comment by mboersma** [comment]
@jadarsie should we merge this now since #2987 has already merged? 🎉

**Comment by jackfrancis** [comment]
Resultant work merged in #2987

**Comment by jackfrancis** [comment]
Yeah, let's merge this as a historical document of the initial design intent

**Pull request merged**